examples/heat-source-mpi.py

```diff
@@ -34,7 +34,7 @@
 from grudge import sym as grudge_sym
 from grudge.shortcuts import make_visualizer
 from grudge.symbolic.primitives import QTAG_NONE
-from mirgecom.integrators import rk4_step
+from mirgecom.timesteppers import RK4Classical
```

Comment:
Ah, this is a very good point! Having `steppers` and `timesteppers` is indeed confusing. I will revert back to `integrators` to keep it more easily distinguishable.

As for where to put `advance_state`, I think putting it in `simutil.py` could work. What do others think? @inducer @MTCam ?

Reply:
I'm not opposed to moving `advance_state`, but I am not sure `simutil.py` is where it should land.  Is there any reason it shouldn't get dropped into the same module as the time integration methods?

Reply:
@MTCam I just took a gander in `simutil.py` and I'm inclined to agree with you; it seems to be a bit out of place compared to the other utility functions in there, and given the role of `advance_state`, I think it makes sense to move it into the timestepping module. Will update!

Reply:
> I'm not opposed to moving `advance_state`, but I am not sure `simutil.py` is where it should land.

Why not? It's a driver helper function like everything else in there.

> Is there any reason it shouldn't get dropped into the same module as the time integration methods?

`advance_state` sits at a higher level in the mirgecom abstraction hierarchy. I don't think it would make sense to package it together with the integrators.

Reply:
> > I'm not opposed to moving `advance_state`, but I am not sure `simutil.py` is where it should land.
> 
> Why not? It's a driver helper function like everything else in there.
> 
> 
> `advance_state` sits at a higher level in the mirgecom abstraction hierarchy. I don't think it would make sense to package it together with the integrators.

I don't disagree with anything you've said here.  My only reluctance to drop it into `simutils.py` is that placing it there would be an _elevation_ of a take-it-or-leave-it, "Random collection-o-tools" as it is documented, module to a higher "status" so-to-say.

Reply:
> I don't disagree with anything you've said here. My only reluctance to drop it into `simutils.py` is that placing it there would be an _elevation_ of a take-it-or-leave-it, "Random collection-o-tools" as it is documented, module to a higher "status" so-to-say.

That's fair. I think if we want to avoid elevating `simutils.py` to that status my preference would be to leave things as they are now with `steppers.py` and `integrators.py`.

Reply:
Sorry I just now saw this discussion. I moved `advance_state` to the time-stepping (now called `integrators`) module in the file `tsutils.py` (similar to the style `simutils.py`). I don't mind keeping it `steppers.py` if that is preferable. But
```python
from mirgecom.integrators import *
```
will include `advance_state` now, as well as all the timestepping functions.

Reply:
Yeah, personally I think it'd be best to keep `advance_state` in `steppers`, and let `integrators` remain blissfully unaware that things like drivers, logging, etc. exist.